.github/workflows/ci.yml

```diff
@@ -27,26 +27,24 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
 
-      - name: Setup python
-        uses: actions/setup-python@v5
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v5
         with:
           python-version: ${{ matrix.python-version }}
 
       - name: Test Installation
         shell: bash
         run: |
-          pip install -r dev-requirements/build.txt
+          uv pip install .
+          uv pip uninstall hikari
 
-          pip install .
-          pip uninstall -y hikari
-
-          pip install .[speedups]
-          pip uninstall -y hikari
+          uv pip install .[speedups]
+          uv pip uninstall hikari
 
       - name: Run tests
         shell: bash
         run: |
-          pip install -r dev-requirements.txt
+          uv pip install --group nox
```

Comment:
Not sure if `uv pip` works with `uv.lock`

Reply:
Also (discussed in DM) you should use `--frozen` with `uv sync` when you switch this over, so the lockfile doesn't update in CI. Even if you don't commit the lockfile back, probably still good to include